frontend/packages/pipelines-plugin/src/components/shared/workspaces/VolumeClaimTemplateLink.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { PersistentVolumeClaimKind } from '@console/internal/module/k8s';
+import { PersistentVolumeClaimModel } from '@console/internal/models';
+import { ResourceLink } from '@console/internal/components/utils';
+import { PipelineRunModel } from '../../../models';
+import { getMatchedPVCs } from '../../../utils/pipeline-utils';
+
+export interface VolumeClaimTemplatesLinkProps {
+  namespace: string;
+  ownerResourceName: string;
+  ownerResourceKind: string;
+}
+
+const VolumeClaimTemplatesLink: React.FC<VolumeClaimTemplatesLinkProps> = ({
+  namespace,
+  ownerResourceName,
+  ownerResourceKind = PipelineRunModel.kind,
+}) => {
+  const [pvcResources, loaded, loadError] = useK8sWatchResource<PersistentVolumeClaimKind[]>({
+    kind: PersistentVolumeClaimModel.kind,
+    isList: true,
+    namespace,
+  });
+
+  if (!ownerResourceName) return null;
+
+  if (!loaded || loadError || pvcResources?.length === 0) return null;
+
+  const matchedPVCs = getMatchedPVCs(pvcResources, ownerResourceName, ownerResourceKind);
+
+  if (!matchedPVCs || matchedPVCs.length === 0) return null;
+
+  return (
+    <>
+      {matchedPVCs.map((pvcResource) => {
+        return (
+          <ResourceLink
+            name={pvcResource.metadata.name}
+            key={pvcResource.metadata.name}
+            namespace={namespace}
```

Comment:
Nit: Technically this should be `pvcResource.metadata.namespace` -- the logic around this makes the two match but your namespace needs to be that of the `pvcResource` not of your PLR/TR